src/cpp/shared/adventure/adv.h

```diff
@@ -66,34 +67,30 @@ class armyGroup {
 
 class hero {
 public:
-  __int16 spellpoints;
+  short spellpoints;
   char idx;
   char ownerIdx;
-  char field_4;
-  char field_5;
-  char field_6;
-  char field_7;
-  char field_8;
-  char field_9;
-  char name[13];
+  short aiLastHeroInteractionTurn;
+  char aiLastHeroInteractionIdx;
+  short aiLastTownInteractionTurn;
+  char aiLastTownInteractionIdx;
+  char name[MAX_HERO_NAME_LENGTH];
   char factionID;
   char heroID;
   int x;
   int y;
-  __int16 field_21;
-  __int16 field_23;
-  __int16 field_25;
-  __int16 field_27;
-  __int8 relatedToX;
-  __int8 relatedToY;
-  __int8 relatedToFactionID;
-  __int8 directionFacing;
-  __int16 occupiedObjType;
-  __int16 occupiedObjVal;
+  int aiTargetX;
```

Comment:
These are 32-bit ints? For an x-coordinate? Any idea why? Usually map coordinates are smaller.